pipefunc/_pipeline.py

```diff
@@ -1125,6 +1132,39 @@ def copy(self) -> Pipeline:
             cache_kwargs=self._cache_kwargs,
         )
 
+    def _independent_parameters(self: Pipeline) -> list[set[str]]:
+        """Return the sets of input and output parameters that are independent."""
+        # Note: could use `_connected_components` instead of `join_overlapping_sets`
+        # Not used ATM...
+        sets = [set(f.parameters) | set(at_least_tuple(f.output_name)) for f in self.functions]
+        return join_overlapping_sets(sets)
+
+    def _connected_components(self) -> list[set[PipeFunc | str]]:
+        """Return the connected components of the pipeline graph."""
+        return list(nx.connected_components(self.graph.to_undirected()))
+
+    def _group_functions_by_chains(self: Pipeline) -> list[set[PipeFunc]]:
+        """Group functions by independent chains / disconnected subgraphs."""
+        connected_components = self._connected_components()
+        return [{x for x in xs if isinstance(x, PipeFunc)} for xs in connected_components]
+
+    def _independent_axes_in_mapspecs(self: Pipeline) -> list[tuple[set[PipeFunc], set[str]]]:
```

Comment:
I realized that one only needs to check the indices of the outputs and then follow those outputs down the graph making sure they still exist at the end.